Python doesn't directly effortlessly support the use of a PGO profile that
wasn't generated during the *current* build.

profile-opt includes PGO profile generation, which is slow (requires multiple
builds & a lengthy test phase), and makes determinism harder. If we place a PGO
profile in the place where profile-opt normally puts one & instead use
build_all_use_profile, all of this works effortlessly with a pre-rolled
profile.

--- a/configure.ac
+++ b/configure.ac
@@ -1299,7 +1299,7 @@ if test "$Py_OPT" = 'true' ; then
   # compile working code using it and both test_distutils and test_gdb are
   # broken when you do manage to get a toolchain that works with it.  People
   # who want LTO need to use --with-lto themselves.
-  DEF_MAKE_ALL_RULE="profile-opt"
+  DEF_MAKE_ALL_RULE="build_all_use_profile"
   REQUIRE_PGO="yes"
   DEF_MAKE_RULE="build_all"
 else
